java/arcs/core/storage/StorageProxy.kt

```diff
@@ -77,7 +77,7 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperation, T>(
             syncMutex.withLock { isSynchronized to crdt.versionMap.copy() }
 
         if (firstReader) requestSynchronization()
-        else if (hasSynced) coroutineScope { launch { handle.callback?.onSync() } }
+        else if (hasSynced) coroutineScope { launch { handle.onSync(versionMap.copy()) } }
```

Comment:
Do we actually need to copy here? versionMap itself is a copy already.

Reply:
That copy will be returned from `registerHandle`, and could be theoretically modified by the caller.

But while we are here: What's the use case for returning a `VersionMap` here? It doesn't seem that this value actually gets used anywhere, currently.

Reply:
(In any case, the change below means there's no need for a copy here, now)

Reply:
Great question, this was one of the things I gave up on asking why and just ported from TS. With this change now the handle should be getting an onsync quickly after registration if the proxy is synced.
I am thinking we should probably set the current version map onto the handle even if we are not synced.
Either way, returning versionmap is not very useful and I agree we can remove it.